### PR TITLE
Rename kyuubi-extension-spark_3.1 to kyuubi-extension-spark-3-1

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

`kyuubi-extension-spark_3.1` is conflict with the scala naming convention, e.g. `spark-core_2.12`

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: dev/kyuubi-extension-spark-3-1/pom.xml [hunk 1]
```diff
@@ -27,9 +27,9 @@
     </parent>
     <modelVersion>4.0.0</modelVersion>
 
-    <artifactId>kyuubi-extension-spark_3.1</artifactId>
+    <artifactId>kyuubi-extension-spark-3-1</artifactId>
     <packaging>jar</packaging>
-    <name>Kyuubi Project Dev Spark Extensions</name>
+    <name>Kyuubi Project Dev Spark Extensions (for Spark 3.1)</name>
```
Comment (pan3793) [inline dev/kyuubi-extension-spark-3-1/pom.xml:32]: It should work with Spark 3.2, we can update it after Spark 3.2 release.

File: dev/kyuubi-extension-spark-3-1/pom.xml [hunk 2]
```diff
@@ -27,9 +27,9 @@
     </parent>
     <modelVersion>4.0.0</modelVersion>
 
-    <artifactId>kyuubi-extension-spark_3.1</artifactId>
+    <artifactId>kyuubi-extension-spark-3-1</artifactId>
```
Comment (pan3793) [inline dev/kyuubi-extension-spark-3-1/pom.xml:30]: Name the module by minimal compatible version, e.g. 
`elasticsearch-spark-13` Elasticsearch Spark (for Spark 1.3 1.6)
`elasticsearch-spark-20` Elasticsearch Spark (for Spark 2.X)

Comment (ulysses-you) [reply dev/kyuubi-extension-spark-3-1/pom.xml:30]: `spark-3-1` seems strange, `spark-3_1` ? or just `spark_31`

Comment (pan3793) [reply dev/kyuubi-extension-spark-3-1/pom.xml:30]: we should avoid using `_`.`3-1` match the style of `spark-sql-kafka-0-10`, I'm fine with both `3-1` and `31`

**Review by ulysses-you — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master for v1.3.0

**Pull request closed without merging**